ml-fundamentals/ml-feasibility-study.md

```diff
@@ -0,0 +1,98 @@
+
+## Who collaborates on ML feasibility studies?
+
+Collaboration from individuals with diverse skillsets is desired at this stage, including data scientists, data engineers, software engineers and PMs. It embraces the use of engineering fundamentals, with some flexibility. For example, not all experimentation requires full test coverage and code review. Experimentation is typically not part of a CI/CD pipeline. Artifacts may live in the master branch as a folder excluded from the CI/CD pipeline, or as a separate experimental branch, depending on customer/team preferences.
+
+## What do ML feasibility studies entail?
+
+### ML problem definition and desired outcome
+
+* Ensuring that the problem is complex enough that coding rules or manual scaling is unrealistic
+* Clear definition of the problem from the ML perspective
+* Definition of precisely what will the ML component solve
+
+### Data discovery
+
+* Hold a data exploration workshop and deep dive with domain experts
+* Understand the data dictionary, data quality and access (availability)
+* Ensure required data is present in reasonable volumes
+* For supervised problems (most common), assess the availability of labels or data that can be used to effectively approximate labels
+* If applicable, ensure all data can be joined as required and understand how
+  * Ideally obtain or create an entity relationship diagram (ERD)
+* Potentially uncover new useful data sources
+
+### Architecture discovery
+
+* Clear picture of existing architecture
+* Infrastructure spikes
+
+### Exploratory data analysis (EDA)
+
+* Data deep dive
+* Understand feature value distributions
+* Understand feature correlations
+* Understand data specific problem constraints like missing values, categorical cardinality, potential for data leakage etc.
+* Identify any gaps in data that couldn’t be identified in the data discovery phase
+* Pave the way of further understanding of what techniques are applicable
+* Establish a mutual understanding of what data is in or out of scope for feasibility, ensuring that the data in scope is significant for the business
+
+### Data pre-processing
+
+* Happens during EDA and hypothesis testing
+* Feature engineering
+* Sampling
+* Scaling and/or discretization
+* Noise handling
+
+### Hypothesis testing
+
+* Design several potential solutions using theoretically applicable algorithms and techniques
```

Comment:
Maybe we should also propose to come up with a baseline (rule based or a simple model)

Reply:
Extended the first point as follows:
Design several potential solutions using theoretically applicable algorithms and techniques, starting with the simplest reasonable baseline